### ISSUE TITLE
Investigate adding end-to-end testing on the portal

### ISSUE THREAD

**Issue by edpin** [tenchi-security/zanshin_status#13]
Using a blackbox testing suite such as cypress.io or Selenium, we should do at least a minimal test to validate that a user can log in, list its scan targets, check (daily or hourly) that the scans have happened, etc. We should alert when this simple test fails.

**Comment by spicer-tenchi** [comment]
@edpin Is this task to "Investigate" or get the basic stuff in place as mentioned above. If the plan is to implement as part of this task I am thinking this will flush into next sprint.

**Comment by edpin** [comment]
This is more to make a plan. When we have a plan, we can add a task list here and possibly create separate tasks to track them. They will likely be in future sprints, yes.

**Comment by spicer-tenchi** [comment]
Ok here are my findings and thoughts. 

I am a big fan of cypress.io. I have used it a bit in the past and it works well. I poked around at anything I could google and everything seemed to be like cypress.io but not free. cypress.io is free but they have some paid features we might want but the pricing seems reasonable compared to others. 

Selenium / Web driver is good too but a little more low level than cypress. Not sure but I would not be surprised if cypress was built on Selenium. I think raw Selenium would work but we get more with cypress. 

The advantage of some of the cypress competitors is many of them have a no code option. More or less you record flows and then they can be tested over and over again. I have not played with this much but I have worked with some shops in the past that use this. They hire a QA team that are not coders, and as they QA they build these E2E tests. I found this strategy to work well. Sort of off loading some of the E2E testing requirements of the developers and making it so the QA team writes the tests. Does not sound like we are going to hire a non-coding QA team anytime soon but this approach is something to consider. If this option is considered I think we should play more with the non-coding tools. 

I am very excited to get these E2E testing in place. I think we have a lot of work ahead of us to get good coverage in the portal via unit tests. These E2E tests would super charge getting better "coverage". 

We could spin up a separate stack (api / engine) to run these tests against. My early thought is back burner that for now and just run these tests against our dev env. Maybe create a handful of special organizations that we test against (so humans do not come in and mess up these organizations and break our tests). 

I have spent some time revisiting cypress and so far I don't see anything that is a show stopper. I think we could get it up and running with basic tests pretty quickly. Even add it to our github actions.

Unless there is better ideas I say we start running with cypress.
